Welcome-Tab feature branch

jshreyans551 opened the pull request.

walterbender commented:
A few outstanding issues:

(1) Skip should be replaced by [x]
(2) Since this code is also used for individual block help, the arrows are not always need.
(3) When this code is also used for individual block help, the title should be the block name.

jshreyans551 commented:
I'm working on the first point
However, could you please explain what you mean in the second and third points. Thanks!

walterbender commented:
We use the help widget for showing help for individual blocks. (You access this with a right-click on a block and then clicking on the ?)

jshreyans551 commented:
Yeah I got what you're saying. I'll add the necessary features by the earliest.

walterbender commented:
This version is looking good. Let me know if you need help with the "help" (vs "tour") functionality.

jshreyans551 commented:
Great!
I was going to ask you the same thing. I've been working on the help functionality but it seems I'm working with the wrong files because any change I make results in unexpected errors. Can you point to the relevant files?

walterbender commented:
The correct file is https://github.com/sugarlabs/musicblocks/blob/master/js/widgets/help.js
It is launched from several different places:
https://github.com/sugarlabs/musicblocks/blob/master/js/activity.js#L3088
https://github.com/sugarlabs/musicblocks/blob/master/js/activity.js#L3096
https://github.com/sugarlabs/musicblocks/blob/master/js/activity.js#L3776
https://github.com/sugarlabs/musicblocks/blob/master/js/block.js#L5683

jshreyans551 commented:
I'm really sorry for the delay. I think I'm familiar with the starter code for the welcome tab now. I'll try to upload changes by today evening.

jshreyans551 commented:
I can't see why the help functionality won't work. I've not touched the code pertaining to it and it still shows the welcome tab content every time. Could you help out?
Once I get the help content rendered, the left/right buttons and the heading won't be a problem

walterbender commented:
I will take a look at your latest patch...

walterbender commented:
I am see an error on Line 78 trying to insert something in the row (which does not exist).
   Uncaught ReferenceError: row is not defined

 In the original code, we used a table. See https://github.com/sugarlabs/musicblocks/blob/master/js/widgets/help.js#L47

walterbender commented:
Also, url(" + '../musicblocks/header-icons/close.png' + ")" doesn't seem to work.
I think you want 'header-icons/close.png'  (Same with drag).

jshreyans551 commented:
@walterbender finally done with the work!
Please forgive the delay. This was the first time I have worked with such a large project and the code was a little overwhelming at first.

> Also, url(" + '../musicblocks/header-icons/close.png' + ")" doesn't seem to work.
Strange, it works for me when testing locally. Still I'll have a look again.

jshreyans551 commented:
However, there are a couple of issues that I think could prove important:
1. The positioning of the text and images when rendering help content is a little off.
2. Now since the title of each help block is given much more significance visually, some of the titles look strange. For eg: the print block looks strange with the title 'print' and some blocks don't even have titles:

![1](https://user-images.githubusercontent.com/41480918/52901265-5dc7cf80-3227-11e9-8bb4-e8966be34cfb.PNG)
![capture](https://user-images.githubusercontent.com/41480918/52901266-5ef8fc80-3227-11e9-9905-47e91ad37ddb.PNG)

Any ideas?

walterbender commented:
I'll take a look. I am a bit surprised that some blocks are missing titles as every block has a label with the exception of number and text blocks.

walterbender commented:
The icon paths in help.js are still wrong. Should not be ../musicblocks/header-icons/...  just header-icons/...

walterbender commented:
Also, please use sans-serif instead of roboto

jshreyans551 commented:
> I am a bit surprised that some blocks are missing titles as every block has a label with the exception of number and text blocks.

Oh well then its all right because the screenshot is from a print text block



> The icon paths in help.js are still wrong. Should not be ../musicblocks/header-icons/... just header-icons/...

Corrected. I've moved the styling code to the activity.css file itself. Hope its correct this time.
The font had also been changed. There was a bug that once the left/right arrows were removed for the help block, they didn't show up again when the tour function was used. That's resolved now.

walterbender commented:
topDiv.innerHTML = "Take a Tour";

Needs to be set up for i18n: _('Take a tour');

jshreyans551 commented:
I've been reading up on how i18n works (wasn't previously aware of it) but I still can't seem to be able to implement it here. Could you help out?

walterbender commented:
There is a function in utils.js names _ which will convert a string to its localized version. So simply wrap your text inside a function call:

instead of "Take a tour" use _("Take a tour")

jshreyans551 commented:
> instead of "Take a tour" use _("Take a tour")

This is done

I guess the individual headings do not need to be setup ( the ones used with the help functionality ) ? Because they're already being used as a variable and not hard-coded text.

walterbender commented:
@pikurasa can you provide me a translation for "Take a tour"? I don't want this to go live without the Kanji and Kana strings in place. Thx

walterbender commented:
merged by hand

jshreyans551 commented:
Thanks!
